operator/auth/spire/client.go

```diff
@@ -135,6 +147,14 @@ func (c *Client) connect(ctx context.Context) (*grpc.ClientConn, error) {
 	timeoutCtx, cancelFunc := context.WithTimeout(ctx, c.cfg.SpireServerConnectionTimeout)
 	defer cancelFunc()
 
+	resolvedTarget, err := resolvedK8sService(ctx, c.k8sClient, c.cfg.SpireServerAddress)
```

Comment:
This function should only return an error when there is a genuine problem. In other words, only log if the supplied domain name is of the format `<foo>.<bar>.svc`, and we fail to look it up because the apiserver goes to bed or the name isn't found.

Reply:
thanks, this should be addressed along with other comments.